packages/twenty-front/src/pages/settings/security/SettingsSecurityTrustedDomain.tsx

```diff
@@ -0,0 +1,141 @@
+import { SaveAndCancelButtons } from '@/settings/components/SaveAndCancelButtons/SaveAndCancelButtons';
+import { SettingsPath } from '@/types/SettingsPath';
+import { SnackBarVariant } from '@/ui/feedback/snack-bar-manager/components/SnackBar';
+import { useSnackBar } from '@/ui/feedback/snack-bar-manager/hooks/useSnackBar';
+import { SubMenuTopBarContainer } from '@/ui/layout/page/components/SubMenuTopBarContainer';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Controller, useForm } from 'react-hook-form';
+import { useNavigateSettings } from '~/hooks/useNavigateSettings';
+import { getSettingsPath } from '~/utils/navigation/getSettingsPath';
+import { domainSchema } from '@/settings/security/validation-schemas/domainSchema';
+import { Trans, useLingui } from '@lingui/react/macro';
+import { TextInput } from '@/ui/input/components/TextInput';
+import { z } from 'zod';
+import { H2Title, Section } from 'twenty-ui';
+import { SettingsPageContainer } from '@/settings/components/SettingsPageContainer';
+import { useCreateWorkspaceTrustDomainMutation } from '~/generated/graphql';
+
+export const SettingsSecurityTrustedDomain = () => {
+  const navigate = useNavigateSettings();
+
+  const { t } = useLingui();
+
+  const { enqueueSnackBar } = useSnackBar();
+
+  const [createWorkspaceTrustDomain] = useCreateWorkspaceTrustDomainMutation();
+
+  const formConfig = useForm<{ domain: string; email: string }>({
+    mode: 'onChange',
+    resolver: zodResolver(
+      z
+        .object({
+          domain: domainSchema,
+          email: z.string().min(1),
+        })
+        .strict(),
+    ),
+    defaultValues: {
+      email: '',
+      domain: '',
+    },
+  });
+
+  const domain = formConfig.watch('domain');
+
+  const handleSave = async () => {
+    try {
+      createWorkspaceTrustDomain({
+        variables: {
+          input: {
+            domain: formConfig.getValues('domain'),
+            email:
+              formConfig.getValues('email') +
+              '@' +
+              formConfig.getValues('domain'),
+          },
+        },
+        onCompleted: () => {
+          enqueueSnackBar(t`Domain added successfully.`, {
+            variant: SnackBarVariant.Success,
+          });
+          navigate(SettingsPath.Security);
+        },
+        onError: (error) => {
+          enqueueSnackBar((error as Error).message, {
+            variant: SnackBarVariant.Error,
+          });
+        },
+      });
+    } catch (error) {
+      enqueueSnackBar((error as Error).message, {
+        variant: SnackBarVariant.Error,
+      });
+    }
+  };
+
+  return (
+    <SubMenuTopBarContainer
+      title="New Trusted Email Domain"
+      actionButton={
+        <SaveAndCancelButtons
+          isSaveDisabled={!formConfig.formState.isValid}
```

Comment:
I dislike the forms where they disable save but don't tell you why, can we provide feedback on save?

Reply:
I improved this part